rasterio/rio/warp.py

```diff
@@ -135,8 +135,7 @@ def warp(
                         raise click.BadParameter('Required when using --bounds',
                             param='res', param_hint='res')
 
-                    xmin, ymin, xmax, ymax = transform_bounds(src.crs, dst_crs,
-                                                              *bounds)
+                    xmin, ymin, xmax, ymax = bounds
```

Comment:
It was my intent when I wrote this that these were in the source coordinate system, and documented that way in the docstring and in `docs/cli.rst`.  Whether or not that was the right choice could certainly be debated, so don't let me hold you back from changing that.

The problem, I think, came about through consolidation / reuse of the `--bounds` option, which is used more generally, and specifically says 'Output bounds: ...'